### PR TITLE
Use high resolution timer on PHP 7.3+ 

### LINKED ISSUE #73: Use monotonic time when available
Currently, PHP doesn't support and built-in way to access the system's monotonic time. That might change in PHP 7.2. I plan to implement a `hrtime()` function providing access to the high-resolution monotonic system time. If UV is available, but another loop is used, `uv_hrtime()` could be used in the meantime.

This issue exists to keep track of any improvements in PHP itself and to not forget implementing `hrtime()` in the future.

~~~https://externals.io/thread/592#email-14248~~~ updated: https://externals.io/message/97502

### PR DESCRIPTION
**Pull request by clue**
This project suggests that event loop implementations SHOULD use a
monotonic time source if available. Given that a monotonic time source is
only available as of PHP 7.3 by default, event loop implementations MAY
fall back to using wall-clock time. http://php.net/manual/en/function.hrtime.php

Builds on top of #157 and #130
Resolves #73

### CODE REVIEW

File: src/Timer/Timers.php [hunk 1]
```diff
@@ -18,10 +18,17 @@ final class Timers
...
+    public function __construct()
+    {
+        // prefer high-resolution timer, available as of PHP 7.3+
+        $this->useHighResolution = \function_exists('hrtime');
+    }
 
     public function updateTime()
     {
-        return $this->time = \microtime(true);
+        return $this->time = $this->useHighResolution ? \hrtime(true) * 1e-9 : \microtime(true);
```
Comment (WyriHaximus) [inline src/Timer/Timers.php]: > `* 1e-9`

What is the reasoning for doing that?

Comment (CharlotteDunois) [reply src/Timer/Timers.php]: On 64bit `hrtime` returns an integer - on 32bit a float. So the idea is to get a float just like `microtime`, but on 32bit this will break the timer.

Comment (WyriHaximus) [reply src/Timer/Timers.php]: Is there a way we can test this on both 64 and 32 bit on Travis? Just to be sure we don't break anything for our users

Comment (clue) [reply src/Timer/Timers.php]: This converts from nanoseconds (which could be given an `int` or `float` depending on architecture) to a `float` second value (`1e-9` means `10^-9` because 1s = 10^9ns). Even on 32 bit platforms, a `float` value in PHP usually follows the 64 bit IEEE format. (http://php.net/manual/en/language.types.float.php and https://stackoverflow.com/questions/1101463/the-double-byte-size-in-32-bit-and-64-bit-os).

In other words: We have no reason to believe that this wouldn't work on any system that this project currently works on. We already use `float` values to store the timestamps in other places in this very project.

Comment (WyriHaximus) [reply src/Timer/Timers.php]: Thank you for that explanation @clue :+1:
